src/Form/Form.js

```diff
@@ -444,7 +452,9 @@ class Form {
 
     get submitButtons () {
         const selector = this.matching.cssSelector('submitButtonSelector')
-        const allButtons = /** @type {HTMLElement[]} */([...this.form.querySelectorAll(selector)])
+        const allButtons = /** @type {HTMLElement[]} */([
+            ...findEnclosedElements(this.form, selector)
+        ])
```

Comment:
No need to spread here now :)

```suggestion
          const allButtons = /** @type {HTMLElement[]} */(findEnclosedElements(this.form, selector))
```

Reply:
Thanks and sorry for leaving it out, almost played with the types moved the functions too many times and lost track :D

Reply:
yeah the Element vs HTMLElement thing is annoying, there are a few ways to improve the ergonomics of it, we discuss another time :)